lab-jaren/server.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+const net = require('net');
+const server = net.createServer();
+const parse = require('./lib/parse-message.js');
+
+let sockets = [];
+
+server.on('connection', function(socket){
+  console.log('a client connected');
+  socket.write('welcome to wack chat\n');
+  socket.username = `user_${Math.floor(Math.random() * 100)}`;
+  sockets.push(socket);
+
+  socket.on('data', function(buffer){
+    console.log(buffer.toString());
+    let message = buffer.toString();
+    if(message.startsWith('/nick'))
+      return parse.nickCommand(message, socket);
+
+    if(message.startsWith('/dm'))
+      return parse.dmCommand(message, sockets);
+
+    if(message.startsWith('/users'))
+      return parse.usersCommand(sockets, socket);
+
+    if(message.startsWith('/troll'))
+      return parse.trollCommand(message, sockets);
+
+    if(message.startsWith('/ban'))
+      return parse.banCommand(message, sockets);
+
+    if(message.startsWith('/info'))
+      return parse.infoCommand(socket);
+
+    sockets.forEach(s => {
```

Comment:
Not a problem re: points, but it might be good to take this loop out because you're broadcasting everything you type to all users by default, which isn't the best UI experience.